### PR TITLE
[AI] Fix pylint warnings

### PR DESCRIPTION
**Pull request by Cjkjvfnby**
- Check if division is true or floor
- Make floor division explicit
- Remove some unused methods (class still have ~900 lines)
- Fix argument for constructor of BasicStats, because `initialMeterValue` always return float

### CODE REVIEW

File: default/python/AI/CombatRatingsAI.py [hunk 1]
```diff
@@ -148,7 +149,7 @@ def __get_stats_from_ship(self):
                         # TODO: Depending on future implementation, might actually need to handle this case.
                         warn("Multiple hangar types present on one ship, estimates expected to be wrong.")
                     fighter_damage = max(fighter_damage, part_damage)
-        self._basic_stats = self.BasicStats(attacks, structure, shields)
+        self._basic_stats = self.BasicStats(attacks, int(structure), int(shields))
```
Comment (Morlic-fo) [inline default/python/AI/CombatRatingsAI.py]: should fix these in the constructor to be ```float``` instead considering they are floats in the game engine.

File: default/python/AI/ShipDesignAI.py [hunk 2]
```diff
@@ -1760,7 +1740,7 @@ def _starting_guess(self, available_parts, num_slots):
 
     def _calc_rating_for_name(self):
         self.update_stats(ignore_species=True)
-        return self.design_stats.structure*self._total_dmg()*(1+self.design_stats.shields/10)
+        return self.design_stats.structure*self._total_dmg()*(1+self.design_stats.shields//10)
```
Comment (Morlic-fo) [inline default/python/AI/ShipDesignAI.py]: should be true division.

File: default/python/AI/ShipDesignAI.py [hunk 3]
```diff
@@ -1830,7 +1810,7 @@ def _filling_algorithm(self, available_parts, verbose=False):
         return best_rating, best_partlist
 
     def _calc_rating_for_name(self):
-        base_rating = self.design_stats.structure*self._total_dmg()*(1+self.design_stats.shields/10)
+        base_rating = self.design_stats.structure*self._total_dmg()*(1+self.design_stats.shields//10)
```
Comment (Morlic-fo) [inline default/python/AI/ShipDesignAI.py]: should be true division.

**Review by Morlic-fo — CHANGES_REQUESTED** [review]

File: default/python/AI/CombatRatingsAI.py [hunk 4]
```diff
@@ -52,7 +53,7 @@ def __init__(self, attacks, structure, shields):
             """
 
             :param attacks:
-            :type attacks: dict|None
+            :type attacks: dict[float, int]|None
```
Comment (o01eg) [inline default/python/AI/CombatRatingsAI.py:56]: Is it have sense to use float as a key of dict?

Comment (Morlic-fo) [reply default/python/AI/CombatRatingsAI.py:56]: Yes, the key is the damage of a weapon which is a float in the game engine.

File: default/python/AI/CombatRatingsAI.py (hunk 1 above)
Comment (Cjkjvfnby) [reply default/python/AI/CombatRatingsAI.py]: It is documented as int, so class users expect to get int here.   But I don't have any strong preferences here.

File: default/python/AI/ShipDesignAI.py (hunk 2 above)
Comment (Cjkjvfnby) [reply default/python/AI/ShipDesignAI.py]: My goal was to make code work same as it.  Lets change it to true division.

File: default/python/AI/CombatRatingsAI.py (hunk 1 above)
Comment (Morlic-fo) [reply default/python/AI/CombatRatingsAI.py]: int is a bug.

Comment (Cjkjvfnby) [reply default/python/AI/CombatRatingsAI.py]: I have already pushed commit with the fix

**Pull request merged**